src/GafferOSL/OSLShader.cpp

```diff
@@ -319,6 +333,20 @@ Plug *loadCompoundNumericParameter( const OSLQuery::Parameter *parameter, const
 	/// \todo Get from metadata
 	ValueType minValue( Imath::limits<BaseType>::min() );
 	ValueType maxValue( Imath::limits<BaseType>::max() );
+	if( metadata )
+	{
+		typedef typename TypedParameter<ValueType>::ObjectType ValueDataType;
```

Comment:
> Your assumption is correct, but it's fine to cast to V3fDataBase because V3fData derives from it - the problem only occurs when you're _creating_ a new instance, rather than when you're looking for one. While we're on the subject, I've been thinking it might make sense to ditch GeometricTypedData and move the interpretation onto the PrimitiveVariable class. I haven't given it a great deal of thought though - any obvious concerns?

I _think_ we tried that originally and the problem was various Ops and `dispatchTypedData` use cases that operate on the `GeometricTypedData` directly without any reference to the `PrimitiveVariable`. I thought you had been leaning towards adding it to all `TypedData` with a default `Invalid` value or something?